### ISSUE TITLE
使用pictorialBar，symbolOffset: [0, '-50%'], 当data中第一个值为0的时候，位置出现偏差，显示在左上角

### ISSUE THREAD

**Issue by jkwf** [apache/echarts#12789]
### Version
4.8.0

### Reproduction link
[https://gallery.echartsjs.com/editor.html?c=xYKFsmdRIF&comment=0](https://gallery.echartsjs.com/editor.html?c=xYKFsmdRIF&comment=0)

### Steps to reproduce
// 多系列无法定位
```js
var colors = [{
    type: 'linear',
    x: 0, x2: 1, y: 0, y2: 0,
    colorStops: [{
        offset: 0,
        color: '#27f7fa'
    }, {
        offset: 0.5,
        color: '#27f7fa'
    }, {
        offset: 0.5,
        color: '#23c3e2dd'
    }, {
        offset: 1,
        color: '#23c3e2dd'
    }]
}, {
    type: 'linear',
    x: 0, x2: 1, y: 0, y2: 0,
    colorStops: [{
        offset: 0,
        color: '#7153d3aa'
    }, {
        offset: 0.5,
        color: '#7153d3aa'
    }, {
        offset: 0.5,
        color: '#352a6b'
    }, {
        offset: 1,
        color: '#352a6b'
    }]
}];

var barWidth = 30;

option = {
    title: {
        text: 'Awesome Chart'
    },
    xAxis: {
        data: ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat']
    },
    yAxis: {},
    series: [{
        z:1,
        type: 'bar',
        barWidth:barWidth,
        data:[0, 182, 191, 234, 290, 330, 310],
        itemStyle:{
            normal:{
                color:colors[1]
            }
        },
    },{
        z:2,
        name:'底部',
        type:'pictorialBar',
        data :[1, 1, 1, 1, 1, 1, 1],
        symbol :'diamond',
        symbolOffset:[0,'50%'],
        symbolSize:[barWidth,10],
        itemStyle:{
            normal:{
                color:colors[1]
            }
        },
    },{
        z:3,
        name:'上部1',
        type:'pictorialBar',
        symbolPosition:'end',
        data:[0, 182, 191, 234, 290, 330, 310],
        symbol :'diamond',
        symbolOffset:[0,'-50%'],
        symbolSize:[barWidth - 4,10 * (barWidth - 4) / barWidth],
        itemStyle:{
            normal:{
                borderColor: '#7153d3',
                borderWidth: 2,
                color:'#352a6b'
            }
        },
    }]
};
```
### What is expected?
为0值时，正常显示
When it is 0, it will display normally

### What is actually happening?
位置显示错乱
Disordered position display

**Comment by jkwf** [comment]
<img width="795" alt="截屏2020-06-10 17 32 38" src="https://user-images.githubusercontent.com/17527532/84251862-82dd0180-ab40-11ea-80bc-976b98f9654b.png">

**Comment by jkwf** [comment]
> ### Version
> 4.8.0
> 
> ### Reproduction link
> https://gallery.echartsjs.com/editor.html?c=xYKFsmdRIF&comment=0
> 
> ### Steps to reproduce
> // 多系列无法定位
> 
> var colors = [{
> type: 'linear',
> x: 0, x2: 1, y: 0, y2: 0,
> colorStops: [{
> offset: 0,
> color: '#27f7fa'
> }, {
> offset: 0.5,
> color: '#27f7fa'
> }, {
> offset: 0.5,
> color: '#23c3e2dd'
> }, {
> offset: 1,
> color: '#23c3e2dd'
> }]
> }, {
> type: 'linear',
> x: 0, x2: 1, y: 0, y2: 0,
> colorStops: [{
> offset: 0,
> color: '#7153d3aa'
> }, {
> offset: 0.5,
> color: '#7153d3aa'
> }, {
> offset: 0.5,
> color: '#352a6b'
> }, {
> offset: 1,
> color: '#352a6b'
> }]
> }];
> 
> var barWidth = 30;
> 
> option = {
> title: {
> text: 'Awesome Chart'
> },
> xAxis: {
> data: ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat']
> },
> yAxis: {},
> series: [{
> z:1,
> type: 'bar',
> barWidth:barWidth,
> data:[0, 182, 191, 234, 290, 330, 310],
> itemStyle:{
> normal:{
> color:colors[1]
> }
> },
> },{
> z:2,
> name:'底部',
> type:'pictorialBar',
> data :[1, 1, 1, 1, 1, 1, 1],
> symbol :'diamond',
> symbolOffset:[0,'50%'],
> symbolSize:[barWidth,10],
> itemStyle:{
> normal:{
> color:colors[1]
> }
> },
> },{
> z:3,
> name:'上部1',
> type:'pictorialBar',
> symbolPosition:'end',
> data:[0, 182, 191, 234, 290, 330, 310],
> symbol :'diamond',
> symbolOffset:[0,'-50%'],
> symbolSize:[barWidth - 4,10 * (barWidth - 4) / barWidth],
> itemStyle:{
> normal:{
> borderColor: '#7153d3',
> borderWidth: 2,
> color:'#352a6b'
> }
> },
> }]
> };
> 
> ### What is expected?
> 为0值时，正常显示
> When it is 0, it will display normally
> 
> ### What is actually happening?
> 位置显示错乱
> Disordered position display

if data = [182, 0, 191, 234, 290, 330, 310], it will be correct,but if data = [0, 182, 191, 234, 290, 330, 310],is wrong

**Comment by plainheart** [comment]
Yes. Thank you for reporting! This bug is under discussion in #12793.

**Comment by jkwf** [comment]
> Yes. Thank you for reporting! This bug is under discussion in #12793.

什么时候能有结果呢？
When will there be results?

**Comment by plainheart** [comment]
Fixed in #12793
